src/transformers/models/beit3/modeling_beit3.py

```diff
@@ -0,0 +1,1414 @@
+        incremental_state=None,
+        positions=None,
+        return_dict=None,
+        output_hidden_states=True,
+    ):
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+        if input_ids is None:
+            x = self.vision_embedding(pixel_values, vision_masked_position)
+            encoder_padding_mask = None
+            multiway_split_position = -1
+        elif pixel_values is None:
+            x = self.text_embedding(input_ids)
+            encoder_padding_mask = text_padding_position
+            multiway_split_position = 0
+        else:
+            x1 = self.vision_embedding(pixel_values, vision_masked_position)
+            multiway_split_position = x1.size(1)
+            x2 = self.text_embedding(input_ids)
+            x = torch.cat([x1, x2], dim=1)
+
+            if text_padding_position is not None:
+                encoder_padding_mask = torch.cat(
+                    [
+                        torch.zeros(x1.shape[:-1]).to(x1.device).bool(),
+                        text_padding_position,
+                    ],
+                    dim=1,
+                )
+            else:
+                encoder_padding_mask = None
+        encoder_out = self.encoder(
+            src_tokens=None,
+            encoder_padding_mask=encoder_padding_mask,
+            attention_mask=attention_mask,
+            token_embeddings=x,
+            multiway_split_position=multiway_split_position,
+            incremental_state=incremental_state,
+            positions=positions,
+            return_dict=return_dict,
+        )
+        if not return_dict:
+            encoder_out.append(multiway_split_position)
+        else:
+            encoder_out["multiway_split_position"] = multiway_split_position
+
+        return encoder_out
+
+
+@add_start_docstrings(
+    """Beit3ForVisualReasoning has a MLP head on top of Beit3Model. Beit3 is a multimodal foundation model. The key idea
+     in BEiT-3 is to model images as another language. Beit3 uses multiway Transformers architecture which uses a
+     shared self-attention module.""",
+    BEIT3_START_DOCSTRING,
+)
+class Beit3ForVisualReasoning(Beit3PreTrainedModel):
+    def __init__(self, config):
+        super(Beit3ForVisualReasoning, self).__init__(config)
+        self.beit3 = Beit3Model(config)
+        self.classifier = Beit3MLP(config)
+        self.post_init()
+
+    @add_start_docstrings_to_model_forward(BEIT3_FOR_VISUALREASONING_INPUTS_DOCSTRING)
+    @replace_return_docstrings(output_type=SequenceClassifierOutput, config_class=_CONFIG_FOR_DOC)
+    def forward(
+        self,
+        input_ids,
+        pixel_values,
+        padding_mask,
+        output_hidden_states=None,
+        return_dict=None,
+        labels=None,
+    ):
+        r"""
+        Returns:
+
+        Examples:
+
+        ```python
+        >>> from transformers import Beit3ForVisualReasoning, BeitImageProcessor, Beit3Processor, XLMRobertaTokenizer
```

Comment:
```suggestion
        >>> from transformers import Beit3ForVisualReasoning, Beit3Processor
```
The processor wraps the image processor and tokenizer, so I guess you only need the processor